frontend/src/app/features/in-app-notifications/center/state/ian-center.service.ts

```diff
@@ -0,0 +1,148 @@
+} from 'core-app/core/state/collection-store.type';
+import { InAppNotificationsService } from 'core-app/core/state/in-app-notifications/in-app-notifications.service';
+import {
+  markNotificationsAsRead,
+  notificationsMarkedRead,
+} from 'core-app/core/state/in-app-notifications/in-app-notifications.actions';
+import { InAppNotification } from 'core-app/core/state/in-app-notifications/in-app-notification.model';
+import { IanCenterQuery } from 'core-app/features/in-app-notifications/center/state/ian-center.query';
+import { ID } from '@datorama/akita';
+import {
+  EffectCallback,
+  EffectHandler,
+} from 'core-app/core/state/effects/effect-handler.decorator';
+import { UntilDestroyedMixin } from 'core-app/shared/helpers/angular/until-destroyed.mixin';
+import { Apiv3ListParameters } from 'core-app/core/apiv3/paths/apiv3-list-resource.interface';
+import { ActionsService } from 'core-app/core/state/actions/actions.service';
+import { HalResource } from 'core-app/features/hal/resources/hal-resource';
+import { APIV3Service } from 'core-app/core/apiv3/api-v3.service';
+import { from } from 'rxjs';
+
+@Injectable()
+@EffectHandler
+export class IanCenterService extends UntilDestroyedMixin {
+  readonly id = 'ian-center';
+
+  readonly store = new IanCenterStore();
+
+  readonly query = new IanCenterQuery(this.store, this.ianService);
+
+  activeFacet$ = this.query.select('activeFacet');
+
+  notLoaded$ = this.query.select('notLoaded');
+
+  paramsChanges$ = this
+    .query
+    .select(['params', 'activeFacet']);
+
+  selectNotifications$ = this
+    .paramsChanges$
+    .pipe(
+      switchMap(() => selectCollectionEntities$<InAppNotification>(this.ianService, this.params)),
+      tap((notifications) => this.sideLoadInvolvedWorkPackages(notifications)),
+    );
+
+  aggregatedCenterNotifications$ = this
+    .selectNotifications$
+    .pipe(
+      map((notifications) => (
+        _.groupBy(notifications, (notification) => notification._links.resource?.href || 'none')
+      )),
+    );
+
+  constructor(
+    readonly injector:Injector,
+    readonly ianService:InAppNotificationsService,
+    readonly actions$:ActionsService,
+    readonly apiV3Service:APIV3Service,
+  ) {
+    super();
+  }
+
+  setFacet(facet:InAppNotificationFacet):void {
+    this.store.update({ activeFacet: facet });
+    this.reload();
+  }
+
+  markAsRead(notifications:ID[]):void {
+    this.actions$.dispatch(
+      markNotificationsAsRead({ caller: this, notifications }),
+    );
+  }
+
+  markAllAsRead():void {
+    selectCollection$(this.ianService, this.params)
+      .pipe(
+        take(1),
+      )
+      .subscribe((collection) => {
+        this.markAsRead(collection.ids);
+      });
+  }
+
+  /**
+   * Reload after notifications were successfully marked as read
+   */
+  @EffectCallback(notificationsMarkedRead)
+  private reloadOnNotificationRead(action:ReturnType<typeof notificationsMarkedRead>) {
+    if (action.caller === this) {
```

Comment:
Not sure I like this pattern and if we shouldn't just take the extra request to avoid having to pass around service references through actions. Also, this class has an `id` property, (which I thought would fill in this role) but I'm unsure what it is used for.

Reply:
What if we pass around an origin? In case of the notification center, marking a notification as read will probably have a visible impact on the notifications if they are being reloaded (e.g., new ones could be added). So I think we need some measure to find out where the action stems from.

I agree that it shouldn't be the service instance however

Reply:
Perhaps an origin is better, but I'd even just take the extra request to not have to worry about this at all. In any case, perhaps we can get some syntactic sugar via another decorator that handles the whole origin stuff automatically, so we don't need to remember which actions require this and which don't.